resources/js/Components/CodecheckReviewDisplay.vue

```diff
@@ -0,0 +1,206 @@
+<template>
+  <div class="codecheck-review-display">
+    <h3>{{ t("plugins.generic.codecheck.reviewTitle") }}</h3>
+    
+    <div v-if="submission?.codecheckOptIn" class="codecheck-info">
+      <div class="info-section">
+        <h4>{{ t("plugins.generic.codecheck.status") }}</h4>
+        <span class="status-badge" :class="statusClass">
+          {{ getStatus() }}
+        </span>
+      </div>
+      
+      <div class="info-section" v-if="hasMetadata">
+        <h4>{{ t("plugins.generic.codecheck.identifier.label") }}</h4>
+        <p>{{ metadata.identifier || t("plugins.generic.codecheck.notYetAssigned") }}</p>
+      </div>
+      
+      <div class="info-section" v-if="repositories.length > 0">
+        <h4>{{ t("plugins.generic.codecheck.repositories.title") }}</h4>
+        <ul>
+          <li v-for="(repo, index) in repositories" :key="index">
+            <a :href="repo" target="_blank">{{ repo }}</a>
+          </li>
+        </ul>
+      </div>
+      
+      <div class="info-section" v-if="hasMetadata && metadata.completionTime">
+        <h4>{{ t("plugins.generic.codecheck.completionTime.label") }}</h4>
+        <p>{{ formatDate(metadata.completionTime) }}</p>
+      </div>
+      
+      <div class="info-section" v-if="hasMetadata && metadata.summary">
+        <h4>{{ t("plugins.generic.codecheck.certificate.summary") }}</h4>
+        <p>{{ metadata.summary }}</p>
+      </div>
+      
+      <div class="actions">
+        <pkp-button @click="viewFullMetadata">
+          {{ t("plugins.generic.codecheck.viewFullMetadata") }}
+        </pkp-button>
+      </div>
+    </div>
+    
+    <div v-else class="codecheck-not-opted">
+      <p>{{ t("plugins.generic.codecheck.notOptedIn") }}</p>
+    </div>
+  </div>
+</template>
+
+<script setup>
+import { computed } from 'vue';
+
+const { useLocalize } = pkp.modules.useLocalize;
+const { t } = useLocalize();
+
+const props = defineProps({
+  submission: { type: Object, required: true }
+});
+
+const metadata = computed(() => {
+  if (props.submission.codecheckMetadata) {
+    if (typeof props.submission.codecheckMetadata === 'string') {
+      try {
+        return JSON.parse(props.submission.codecheckMetadata);
+      } catch (e) {
+        return {};
+      }
+    }
+    return props.submission.codecheckMetadata;
+  }
+  return {};
+});
+
+const hasMetadata = computed(() => {
+  return Object.keys(metadata.value).length > 0;
+});
+
+const repositories = computed(() => {
+  if (metadata.value.repositories) {
+    return metadata.value.repositories;
+  }
+  
+  const repos = [];
+  if (props.submission.codeRepository) {
+    repos.push(...props.submission.codeRepository.split('\n').filter(r => r.trim()));
+  }
+  if (props.submission.dataRepository) {
+    repos.push(...props.submission.dataRepository.split('\n').filter(r => r.trim()));
+  }
+  return repos;
+});
+
+const statusClass = computed(() => {
```

Comment:
You are doubling the cases from `getStatus(..)` here; I think it is a good idea though to seperate the internal status name from the CSS class, so instead, let `statusClass` rely on the response of `getStatus()` and translate that to the right class, maybe with a `switch` statement.